emission/individual_tests/TestNominatim.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import *
+import unittest
+import unittest.mock as mock
+# import overpy
+import os
+from emission.core.wrapper.trip_old import Coordinate
+import requests
+import emission.core.wrapper.entry as ecwe
+import emission.core.wrapper.wrapperbase as ecww
+import emission.net.ext_service.geocoder.nominatim as eco
+import emission.analysis.intake.cleaning.clean_and_resample as clean
+
+#temporarily sets NOMINATIM_QUERY_URL to the environment variable for testing.
+NOMINATIM_QUERY_URL_env = os.environ.get("NOMINATIM_QUERY_URL")
+
+NOMINATIM_QUERY_URL = (
+    NOMINATIM_QUERY_URL_env if NOMINATIM_QUERY_URL_env is not None 
+    else eco.NOMINATIM_QUERY_URL
+)
+
+GFBK = os.environ.get("GFBK_KEY")
+
+GEOFABRIK_QUERY_URL = (
+    "https://geocoding.geofabrik.de/{}".format(GFBK) if GFBK is not None
+    else print("No key available")
+)
+
+#Creates a fake place in Rhode Island to use for testing.
+fake_id = "rhodeislander"
+key = "segmentation/raw_place"
+write_ts = 1694344333
+data = {'source': 'FakeTripGenerator','location': {'type': 'Point', 'coordinates': [-71.4128343, 41.8239891]}}
+fake_place = ecwe.Entry.create_fake_entry(fake_id, key, data, write_ts)
+
+class NominatimTest(unittest.TestCase):
+    maxDiff = None
+
+    #Basic query to check that nominatim and geofabrik are calling the same area.
+    def test_geofabrik_and_nominatim(self):
+        nominatim_result = requests.get(NOMINATIM_QUERY_URL + "/reverse?lat=41.8239891&lon=-71.4128343&format=json").json()
+        geofabrik_result = requests.get(GEOFABRIK_QUERY_URL + "/reverse?lat=41.8239891&lon=-71.4128343&format=json").json()
+        key_list = ['osm_id', 'boundingbox']
+        for k in key_list:
+            self.assertEqual(nominatim_result[k], geofabrik_result[k])
+
+    #Checks the display name generated by get_filtered_place in clean_and_resample.py, which creates a cleaned place from the fake place
+    # and reverse geocodes with the coordinates.
+    def test_get_filtered_place(self):
+        raw_result = ecww.WrapperBase.__getattr__(clean.get_filtered_place(fake_place), "data")
```

Comment:
I would suggest using `ecwe.Entry` instead. We don't use `WrapperBase` outside of the `ecw` package, IIRC.

Reply:
Done

Reply:
this is not the way to use ecwe.Entry please look at existing usage patterns and fix